pxf/pxf-api/src/main/java/org/apache/hawq/pxf/api/Metadata.java

```diff
@@ -91,12 +94,17 @@ public Field(String name, EnumHawqType type, String sourceType) {
             this.sourceType = sourceType;
         }
 
-        public Field(String name, EnumHawqType type, String sourceType,
-                String[] modifiers) {
+        public Field(String name, EnumHawqType type, String sourceType, String[] modifiers) {
             this(name, type, sourceType);
             this.modifiers = modifiers;
         }
 
+        public Field(String name, EnumHawqType type, boolean isComplexType, String sourceType, String[] modifiers) {
```

Comment:
Add isComplexType as the final argument to keep the method overloading structure clean